src/fsharp/service/service.fs

```diff
@@ -1214,32 +1214,28 @@ type TypeCheckInfo
                           Some (FSharpFindDeclResult.ExternalDecl (assref.Name, ExternalSymbol.Type fullName))
                       | _ -> None
                   | _ -> None
-       
               match result with
               | Some x -> x
-              | None ->
-                  let fail defaultReason = 
-                      match item.Item with 
+              | None   ->
+              match rangeOfItem g preferFlag item.Item with
+              | Some itemRange -> 
+                  let projectDir = Filename.directoryName (if projectFileName = "" then mainInputFileName else projectFileName)
+                  fileNameOfItem g (Some projectDir) itemRange item.Item
+                  |> mkRange <| itemRange.Start <| itemRange.End              
```

Comment:
Please never use the ``<|`` operator in the compiler code any more than it is today